fontbe/src/features.rs

```diff
@@ -146,10 +482,84 @@ impl Work<Context, Error> for FeatureWork {
         } else {
             debug!("No fea file, dull compile");
         }
+
         // Enables the assumption that if the file exists features were compiled
         if context.flags.contains(Flags::EMIT_IR) {
             fs::write(context.paths.target_file(&WorkId::Features), "1").map_err(Error::IoError)?;
         }
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::{BTreeMap, HashMap, HashSet};
+
+    use fea_rs::compile::VariationInfo;
+    use font_types::{Fixed, Tag};
+    use fontir::{
+        coords::{CoordConverter, DesignCoord, NormalizedCoord, NormalizedLocation, UserCoord},
+        ir::{Axis, StaticMetadata},
+    };
+
+    use super::FeaVariationInfo;
+
+    fn single_axis_norm_loc(tag: Tag, value: f32) -> NormalizedLocation {
+        let mut loc = NormalizedLocation::new();
+        loc.insert(tag, NormalizedCoord::new(value));
+        loc
+    }
+
+    fn weight_variable_static_metadata(min: f32, def: f32, max: f32) -> StaticMetadata {
+        let wght = Tag::new(b"wght");
+        let min_wght_user = UserCoord::new(min);
+        let def_wght_user = UserCoord::new(def);
+        let max_wght_user = UserCoord::new(max);
+        let min_wght = single_axis_norm_loc(wght, -1.0);
+        let def_wght = single_axis_norm_loc(wght, 0.0);
+        let max_wght = single_axis_norm_loc(wght, 1.0);
+        StaticMetadata::new(
+            1024,
+            Default::default(),
+            vec![Axis {
+                name: "Weight".to_string(),
+                tag: Tag::new(b"wght"),
+                min: min_wght_user,
+                default: def_wght_user,
+                max: max_wght_user,
+                hidden: false,
+                converter: CoordConverter::new(
+                    vec![
+                        // the design values don't really matter
+                        (min_wght_user, DesignCoord::new(0.0)),
+                        (def_wght_user, DesignCoord::new(1.0)),
+                        (max_wght_user, DesignCoord::new(2.0)),
+                    ],
+                    1,
+                ),
+            }],
+            Default::default(),
+            Default::default(),
+            HashSet::from([min_wght, def_wght, max_wght]),
+        )
+        .unwrap()
+    }
+
+    #[test]
+    fn compute_kern_at_default() {
+        let wght = Tag::new(b"wght");
+        let static_metadata = weight_variable_static_metadata(300.0, 400.0, 700.0);
+        let var_info = FeaVariationInfo::new(&static_metadata);
+
+        // Find me the default value despite that not being one of the input locations
```

Comment:
this example won't compile in fonttools because, as I said, a value for the baster master (at default location) is always required. You'll probably get "base master not found". What you seem to be doing here is like a naive version of the L4 instancer (moving the default). May seem to make sense for this simple case with one axis and two masters, but will blow up as soon as you make it more complicated. I'll repeat, the general rule in OT variations is that the default is never sparse, always densely contains all the values, it defines the skeleton of the VF, what's left when you take away all the variation tables, it's the reference point for all deltas. The default master cannot be left undefined for it influences all the regions of the designspace (whose deltas are defined in terms of it), therefore the font developer must know exactly and take responsibility to specify where/what that looks like, it should not be left to the compiler to guess.

Reply:
I understand that in the binary it must be dense at 0n. If we differ in opinion it is wrt whether it follows that the _source_ needs to be dense at default. IMHO it doesn't, it should be fine to have a source with say weight 300-700, default 400, with source defining only two outlines, at 300 and 700, for a glyph. 

Ultimately IMHO in source you just need to provide a definition that spans default. That is, provide _at least_ one of:

1. A definition at default
   * if there are no other definitions the glyph won't vary
1. Definition at 2 or more locations that define a region that encompasses 0n
   * compiler will figure out what the value at 0n should be and the deltas to give the expected results